cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2369,6 +2369,856 @@ endmacro()
 # ----------------------------------------------------------------------
 # Dependencies for Arrow Flight RPC
 
+macro(build_absl_once)
+  if(NOT TARGET absl_ep)
+    message(STATUS "Building Abseil-cpp from source")
+    set(ABSL_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/absl_ep-install")
+    set(ABSL_CMAKE_ARGS
+        "${EP_COMMON_CMAKE_ARGS}"
+        -DABSL_RUN_TESTS=OFF
+        -DCMAKE_CXX_STANDARD=11
+        -DCMAKE_INSTALL_LIBDIR=lib
+        "-DCMAKE_INSTALL_PREFIX=${ABSL_PREFIX}")
+    set(ABSL_BUILD_BYPRODUCTS)
+    set(ABSL_LIBRARIES)
+
+    # Abseil produces the following libraries, each is fairly small, but there
```

Comment:
For the record, since most of the declarations below could be generated by a script, perhaps they could be factored out into a separate CMake inclusion file? That would also reduce clutter in this file.